components/common/src/main/java/com/bloxbean/cardano/yaci/store/common/util/GovUtil.java

```diff
@@ -0,0 +1,125 @@
+package com.bloxbean.cardano.yaci.store.common.util;
+
+import com.bloxbean.cardano.client.crypto.Bech32;
+import com.bloxbean.cardano.client.governance.GovId;
+import com.bloxbean.cardano.client.util.HexUtil;
+import lombok.experimental.UtilityClass;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Arrays;
+
+/**
+ * Utility class for governance-related operations.
+ * 
+ * This class provides utilities for CIP-129 compatible governance identifiers
+ * using Cardano Client Library's native support.
+ */
+@Slf4j
+@UtilityClass
+public class GovUtil {
+
+    /**
+     * Converts a CIP-129 compatible gov_action_id bech32 string to transaction hash and index.
+     * 
+     * @param govActionIdBech32 The gov_action_id in bech32 format (e.g., "gov_action1...")
+     * @return GovActionIdParts containing txHash and index
```

Comment:
Let's add a //TODO in the comment. This is temporary implementation, it will be moved to CCL.